### PR TITLE
Read MAX_VALUE_LENGTH from client options (#2121)

### LINKED ISSUE #2121: Introduce config for MAX_STRING_LENGTH
### Problem Statement

In our on-premises Sentry, we're regularly missing some context information (variables, part of variables) in our issues. This is quite annoying since they usually contain ID or hashes that we need to further debug the information.
We want to raise the limit that the Sentry SDK sends. But that limit is [hard coded](https://github.com/getsentry/sentry-python/blob/8672640f60b4c594a18e7c14343c50fe8de34833/sentry_sdk/utils.py#L71).

There is a nice [TODO](https://github.com/getsentry/sentry-python/blob/8672640f60b4c594a18e7c14343c50fe8de34833/sentry_sdk/utils.py#L935), saying it should be read from config, but I cannot find any issues regarding that.

So the only solution would be monkeypatching this value. Which I consider the source of (all) evil and should be avoided, especially in production environment. 

### Solution Brainstorm

1. Introduce a config for MAX_STRING_LENGTH
2. Read MAX_STRING_LENGTH from said config
3. Optional: Document said config with a disclaimer that you can/will break Sentry internals if you raise this limit to a too high value (assumption, I'm sure it's set to it's current value for a reason)

### PR DESCRIPTION
**Pull request by puittenbroek**
Hard to jump through some hoops to get the new client options for `max_string_length` to where it was needed.

Still kept a fallback in place since I'm unsure if client options is always available/passed a long.

Closes #2121

### CODE REVIEW

**Comment by puittenbroek** [comment]
@antonpirker do you have time to review this? Or someone else from your team?

**Comment by sentrivana** [comment]
Hey @puittenbroek, thanks a lot for this PR! We're currently evaluating internally what the potential impact of this change would be with regards to the ingestion pipeline -- we'll get back to you soon.

**Comment by sentrivana** [comment]
@puittenbroek I ran the pipeline in the meantime -- could you please take a look at the failing workflows?

I'll squeeze a review in sometime next week unless Anton beats me to it. We'll also need [docs](https://docs.sentry.io/platforms/python/configuration/options/) ([source](https://github.com/getsentry/sentry-docs/edit/master/src/platforms/common/configuration/options.mdx)) for this where we explicitly warn users that bumping the string limit might result in event payloads getting too big and events getting dropped, but I can take care of writing that.

**Comment by puittenbroek** [comment]
The circular import in apidocs is a PITA, because the consts file imports `from sentry_sdk.integrations import Integration` which itself also imports which need the consts file. Only way around that is putting my new constant at the top of the file.

File: tests/test_exceptiongroup.py [hunk 1]
```diff
@@ -47,6 +47,7 @@ def test_exceptiongroup():
         client_options={
             "include_local_variables": True,
             "include_source_context": True,
+            "max_string_length": 1024,
```
Comment (antonpirker) [inline tests/test_exceptiongroup.py:50]: Why do we need to set those here? Can we not rely on the default value?

File: tests/test_serializer.py [hunk 2]
```diff
@@ -144,3 +144,20 @@ def test_no_trimming_if_request_bodies_is_always(body_normalizer):
...
+
+    result = body_normalizer(data)
+
+    assert len(result["key"]) == 1024  # fallback max length
+
+
+def test_max_string_length(body_normalizer):
+    data = {"key": "a" * 2000}
+
+    max_string_length = 2000
```
Comment (antonpirker) [inline tests/test_serializer.py:160]: Whould we set `max_string_length` to something between the default and 2000 so check that some trimming is happening?

File: tests/test_client.py [hunk 3]
```diff
@@ -1104,3 +1104,18 @@ def test_multiple_positional_args(sentry_init):
...
+    "sdk_options, expected_data_length",
+    [({}, DEFAULT_MAX_STRING_LENGTH), ({"max_string_length": 2000}, 2000)],
+)
+def test_max_string_length_option(
+    sentry_init, capture_events, sdk_options, expected_data_length
+):
+    sentry_init(sdk_options)
+    events = capture_events()
+
+    capture_message("a" * 2000)
```
Comment (antonpirker) [inline tests/test_client.py:1119]: Should we set the length of the string to something bigger than 2000 to make it more clear that some trimming is happening?

File: sentry_sdk/client.py [hunk 4]
```diff
@@ -294,7 +294,10 @@ def _prepare_event(
                     "values": [
                         {
                             "stacktrace": current_stacktrace(
-                                self.options["include_local_variables"]
+                                include_local_variables=self.options[
```
Comment (antonpirker) [inline sentry_sdk/client.py:297]: Please use also `self.options.get(...)` like below.

**Review by antonpirker — COMMENTED** [review]
Have some questions. In general it looks really good!

**Comment by antonpirker** [comment]
Great work @puittenbroek !
I have left a couple of questions/suggestions. 
We then have to fix eventually failing tests and add the new option to the docs repo.

File: tests/test_exceptiongroup.py (hunk 1 above)
Comment (antonpirker) [reply tests/test_exceptiongroup.py:50]: nevermind

**Comment by antonpirker** [comment]
Added docs PR here: https://github.com/getsentry/sentry-docs/pull/7465

**Review by antonpirker — APPROVED** [review]
lgtm

**Comment by puittenbroek** [comment]
> I just discovered that we already have an option "max_value_length" in the PHP sdk, so I will rename the option to be in line with the other SDK.

Makes sense :) And thanks for the docs PR! pretty swamped at work so didn't have much time to pick this up further.

**Comment by antonpirker** [comment]
No problem @puittenbroek !
Thanks again for the contribution, I guess we are now ready to merge this!

**Pull request merged**